proposals/gc/MVP.md

```diff
@@ -173,14 +215,13 @@ This can compile to machine code that (1) reads the RTT from `$x`, (2) checks th
 
 #### Values
 
-* Creating a structure or array optionally allows supplying a suitable RTT value to represent its runtime type; it is `any` if none is given.
+* Creating a structure or array requires supplying a suitable RTT value to represent its runtime type.
 
-* Each reference value has an associated runtime type:
-  - For structures or arrays, it is the RTT value provided upon creation, or `anyref` if none.
-  - For functions, it is the RTT value for the function's type.
-  - For `i31ref` references it is the RTT value for `i31ref`.
+* Reference values of data or function type have an associated runtime type:
+  - for structures or arrays, it is the RTT value provided upon creation,
+  - for functions, it is the RTT value for the function's type.
```

Comment:
Does this mean that functions from the function references proposal would implicitly have `(rtt.canon $t)`? How would one specify another RTT for a function?

Reply:
Yes, for backwards compatibility, functions have a canonical RTT. There was a brief discussion somewhere about a mechanism for picking custom RTTs for functions, but I can't find it anymore. Added a short section to Post-MVP.md. Do you see an immediate use case for it?

Reply:
I am more worried of needing to eagerly compute `rtt.canon` for a function type in `func.ref`, because keeping around enough metadata to lazily compute it might be even worse.

Reply:
Hm, at least uses of `ref.func` need to be pre-declared, so you don't have to that for functions only used second-class. But for the others, what alternative would you suggest? I don't know how we could avoid that first-class functions carry their type information somehow, if we want to allow casts on them.

Reply:
I know, I don't have a good answer right now, other than a vague feeling that `func.ref` and `func.bind` are analogous to allocations that would need explicit RTT values.